Sources/ForageSDK/Foundation/Telemetry/Metrics.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  File.swift
+//  
+//
+//  Created by Danilo Joksimovic on 2023-08-07.
+//
+
+import Foundation
+
+internal func calculateTimeDifferenceInMilliseconds(from startTime: DispatchTime, to endTime: DispatchTime) -> Double {
+    let nanoseconds = endTime.uptimeNanoseconds - startTime.uptimeNanoseconds
+    let milliseconds = Double(nanoseconds) / 1_000_000
+    return milliseconds
+}
+
+internal struct LatencyAttributes {
+    let path: String
+    let method: HttpMethod
+    var httpStatusCode: Int?
+    let roundTripMs: Double
+    let action: VaultAction
+}
+
+internal func logVaultProxyResponseMetrics(vaultType: VaultType, latencyAttributes: LatencyAttributes) {
+    let metricsLogger: ForageLogger? = DatadogLogger(
+        ForageLoggerConfig(
+            prefix: "Metrics"
+        )
+    )
+    let vaultName = vaultType.rawValue
+    metricsLogger?.setLogKind(ForageLogKind.metric).info("Received response from \(vaultName) proxy", attributes: [
+        "vault_type": vaultName,
+        "path": latencyAttributes.path,
+        "method": latencyAttributes.method.rawValue,
+        "http_status": latencyAttributes.httpStatusCode,
+        "roundtrip_ms": latencyAttributes.roundTripMs,
```

Comment:
can we call it latency to standardize on names?

Reply:
Discussed with Shardendu via slack DM

we've agreed on `response_time_ms` [instead of roundtrip or latency](https://docs.google.com/document/d/1ugFFT3Zs1q0I7NRQVdeJVnsYEIBGDtg37iM8PZz10PQ/edit?disco=AAAA4CIpuM8) across all of our vaulting metrics